python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -758,16 +758,24 @@ def _test_rotation(self, src_center, ref_center, src_delta, ref_delta,
         # Now that we have our shift we apply it to the src delta vector
         # and check the rotation.
         rot_src_delta = np.dot(shift_matrix, src_delta)
-        cos_rot_sq = (np.dot(rot_src_delta, ref_delta) ** 2 /
-                      (np.dot(rot_src_delta, rot_src_delta) *
-                       np.dot(ref_delta, ref_delta)))
+        proj_src_delta = (rot_src_delta -
+                          np.dot(rot_src_delta, ref_center) * ref_center)
+        proj_ref_delta = (ref_delta -
+                          np.dot(ref_delta, ref_center) * ref_center)
+        cos_rot_sq = (np.dot(proj_src_delta, proj_ref_delta) ** 2 /
+                      (np.dot(proj_src_delta, proj_src_delta) *
+                       np.dot(proj_ref_delta, proj_ref_delta)))
```

Comment:
I assume that the dot products of vectors with themselves are meant for clarity but you could do np.sum(np.square(x)) instead.

Reply:
Correct, the dots are for clarity and simplicity. I can't say I'm really for making the change as it changes one call into two. I'll leave it as is.